src/app/file-manager/transferables-grid/transferables-grid.component.ts

```diff
@@ -126,21 +130,32 @@ export class TransferablesGridComponent implements OnInit, AfterViewInit {
         }
       });
     });
-    this.gcsService.exportItemCompleted.subscribe(data => {
-      const items = new FilesDatabase(this.store).data
-        .filter(item => item.type === Type.EXPORT_GCP && item.status === ItemStatus.PENDING);
-      if (items.length === 0 && TransferablesGridComponent.isExporting) {
-        this.modeVariable = this.DETERMINATE;
-        this.generalExportToGCPProgress = 100;
-        TransferablesGridComponent.isExporting = false;
-      } else if (TransferablesGridComponent.isExporting) {
-        this.modeVariable = this.INDETERMINATE;
-        this.limitTransferables.exportItems(items);
-      }
-    });
+
+    this.gcsService.exportItemCompleted.subscribe(() => {
```

Comment:
This seems like an odd way for an API user to monitor results from a method call. I would expect the notification method to be passed as a parameter to `exportToGCPFiles()`. The subscribe can be done inside the API implementation, so you don't need to make `exportItemCompleted` part of the abstract API.

Reply:
You're right in this observation, is it ok for you if we create a Jira story for this fix? This approach is introducing some buggy behavior in the UI that we need to handle carefully.